### PR TITLE
[ODS-4409] Moving Swagger Document Generation to use IoC

### PR DESCRIPTION
**Pull request by BhanuprakashThunga**

### CODE REVIEW

File: Application/EdFi.Ods.Features/OpenApiMetadata/Factories/OpenApiMetadataParametersFactory.cs [hunk 1]
```diff
@@ -11,6 +12,12 @@ namespace EdFi.Ods.Features.OpenApiMetadata.Factories
 {
     public class OpenApiMetadataParametersFactory
     {
+        private readonly IDefaultPageSizeLimitProvider _defaultPageSizeLimitProvider;
+
+        public OpenApiMetadataParametersFactory(IDefaultPageSizeLimitProvider defaultPageSizeLimitProvider)
+        {
+            _defaultPageSizeLimitProvider = defaultPageSizeLimitProvider;
+        }
         public IDictionary<string, Parameter> Create(bool isCompositeContext)
```
Comment (jamessmckay) [inline Application/EdFi.Ods.Features/OpenApiMetadata/Factories/OpenApiMetadataParametersFactory.cs:21]: need a blank line between methods

File: Application/EdFi.Ods.Features/OpenApiMetadata/Factories/OpenApiMetadataPathsFactory.cs [hunk 2]
```diff
@@ -65,17 +68,15 @@ public class OpenApiMetadataPathsFactory
                                     Path = $"{resourcePath}/{{id}}",
                                     PathItem = CreatePathItemForAccessByIdsOperations(r)
                                 }
-                                : null
-#if NETFRAMEWORK
-                            ,
-                            ChangeQueryFeature.IsEnabled && !r.Name.Equals(ChangeQueryFeature.SchoolYearTypesResourceName) && !isCompositeContext
+                                : null,
+                            _apiSettings.Features.Where(s => s.Name == "ChangeQueries").SingleOrDefault().IsEnabled &&
```
Comment (jamessmckay) [inline Application/EdFi.Ods.Features/OpenApiMetadata/Factories/OpenApiMetadataPathsFactory.cs:72]: there is a method on apisettings that does the check for the feature IsFeatureEnabled()

File: Application/EdFi.Ods.Features/OpenApiMetadataContentProviders/CompositesOpenApiContentProvider.cs [hunk 3]
```diff
@@ -49,15 +54,17 @@ public IEnumerable<OpenApiContent> GetOpenApiContent()
                         OrganizationCode = x.OrganizationCode,
                         CategoryName = x.Name
                     })
-                .Select(x => new OpenApiMetadataDocumentContext(_resourceModelProvider.GetResourceModel()) { CompositeContext = x })
+                .Select(x => new OpenApiMetadataDocumentContext(_resourceModelProvider.GetResourceModel()) { CompositeContext =
+ x })
```
Comment (jamessmckay) [inline Application/EdFi.Ods.Features/OpenApiMetadataContentProviders/CompositesOpenApiContentProvider.cs:58]: fix formatting

File: Application/EdFi.Ods.Features/OpenApiMetadataContentProviders/CompositesOpenApiContentProvider.cs [hunk 4]
```diff
@@ -40,7 +45,7 @@ public string RouteName
         public IEnumerable<OpenApiContent> GetOpenApiContent()
         {
             var openApiStrategy = new OpenApiCompositeStrategy(_compositesMetadataProvider);
-
+
```
Comment (jamessmckay) [inline Application/EdFi.Ods.Features/OpenApiMetadataContentProviders/CompositesOpenApiContentProvider.cs:48]: remove whitespace

File: Application/EdFi.Ods.Features/OpenApiMetadataContentProviders/EdFiOpenApiContentProvider.net.cs [hunk 5]
```diff
@@ -0,0 +1,56 @@
+﻿// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+#if NETFRAMEWORK
```
Comment (jamessmckay) [inline Application/EdFi.Ods.Features/OpenApiMetadataContentProviders/EdFiOpenApiContentProvider.net.cs:6]: this should be before the namespaces

File: Application/EdFi.Ods.Features/OpenApiMetadataContentProviders/ExtensionsOpenApiContentProvider.cs [hunk 6]
```diff
@@ -24,13 +26,15 @@ public class ExtensionsOpenApiContentProvider : IOpenApiContentProvider
         private readonly IDomainModelProvider _domainModelProvider;
         private readonly IResourceModelProvider _resourceModelProvider;
         private readonly ISchemaNameMapProvider _schemaNameMapProvider;
+        private readonly IOpenApiMetadataDocumentFactory _openApiMetadataDocumentFactory;
 
         public ExtensionsOpenApiContentProvider(IDomainModelProvider domainModelProvider,
-            IResourceModelProvider resourceModelProvider, ISchemaNameMapProvider schemaNameMapProvider)
+            IResourceModelProvider resourceModelProvider, ISchemaNameMapProvider schemaNameMapProvider,IOpenApiMetadataDocumentFactory openApiMetadataDocumentFactory)
```
Comment (jamessmckay) [inline Application/EdFi.Ods.Features/OpenApiMetadataContentProviders/ExtensionsOpenApiContentProvider.cs:32]: fix formatting

File: Application/EdFi.Ods.Tests/EdFi.Ods.Common/Security/EdFiAdminApiClientIdentityProviderTests.cs [hunk 7]
```diff
@@ -45,7 +45,7 @@ public class When_calling_GetApiClientIdentity : TestFixtureBase
             protected override void Arrange()
             {
                 _transaction = new TransactionScope();
-
+
```
Comment (jamessmckay) [inline Application/EdFi.Ods.Tests/EdFi.Ods.Common/Security/EdFiAdminApiClientIdentityProviderTests.cs:48]: remove whitespace

File: Application/EdFi.Ods.Tests/EdFi.Ods.Features/OpenApiMetadata/OpenApiMetadataSdkGenTests.cs [hunk 8]
```diff
@@ -17,26 +18,74 @@
...
 using EdFi.Ods.Tests.EdFi.Ods.Api.Services.Metadata.Helpers;
 using EdFi.TestFixture;
+using Microsoft.Extensions.Configuration;
 using NUnit.Framework;
 using Test.Common;
 using EdFiSchema = EdFi.Ods.Common.Models.Domain.Schema;
 
 namespace EdFi.Ods.Tests.EdFi.Ods.Features.OpenApiMetadata
 {
+
```
Comment (jamessmckay) [inline Application/EdFi.Ods.Tests/EdFi.Ods.Features/OpenApiMetadata/OpenApiMetadataSdkGenTests.cs:28]: remove blank line

**Review by jamessmckay — CHANGES_REQUESTED** [review]

File: Application/EdFi.Ods.Features/OpenApiMetadata/Factories/OpenApiMetadataParametersFactory.cs (hunk 1 above)
Comment (BhanuprakashThunga) [reply Application/EdFi.Ods.Features/OpenApiMetadata/Factories/OpenApiMetadataParametersFactory.cs:22]: Added blank line

File: Application/EdFi.Ods.Features/OpenApiMetadata/Factories/OpenApiMetadataPathsFactory.cs (hunk 2 above)
Comment (BhanuprakashThunga) [reply Application/EdFi.Ods.Features/OpenApiMetadata/Factories/OpenApiMetadataPathsFactory.cs]: Used IsFeatureEnabled method

File: Application/EdFi.Ods.Features/OpenApiMetadataContentProviders/CompositesOpenApiContentProvider.cs (hunk 3 above)
Comment (BhanuprakashThunga) [reply Application/EdFi.Ods.Features/OpenApiMetadataContentProviders/CompositesOpenApiContentProvider.cs]: Formatting issue fixed

File: Application/EdFi.Ods.Features/OpenApiMetadataContentProviders/CompositesOpenApiContentProvider.cs (hunk 4 above)
Comment (BhanuprakashThunga) [reply Application/EdFi.Ods.Features/OpenApiMetadataContentProviders/CompositesOpenApiContentProvider.cs]: Removed whitespace

File: Application/EdFi.Ods.Features/OpenApiMetadataContentProviders/EdFiOpenApiContentProvider.net.cs (hunk 5 above)
Comment (BhanuprakashThunga) [reply Application/EdFi.Ods.Features/OpenApiMetadataContentProviders/EdFiOpenApiContentProvider.net.cs]: Added above namespace

File: Application/EdFi.Ods.Features/OpenApiMetadataContentProviders/ExtensionsOpenApiContentProvider.cs (hunk 6 above)
Comment (BhanuprakashThunga) [reply Application/EdFi.Ods.Features/OpenApiMetadataContentProviders/ExtensionsOpenApiContentProvider.cs]: Formatted

File: Application/EdFi.Ods.Tests/EdFi.Ods.Common/Security/EdFiAdminApiClientIdentityProviderTests.cs (hunk 7 above)
Comment (BhanuprakashThunga) [reply Application/EdFi.Ods.Tests/EdFi.Ods.Common/Security/EdFiAdminApiClientIdentityProviderTests.cs]: Removed whitespace

File: Application/EdFi.Ods.Tests/EdFi.Ods.Features/OpenApiMetadata/OpenApiMetadataSdkGenTests.cs (hunk 8 above)
Comment (BhanuprakashThunga) [reply Application/EdFi.Ods.Tests/EdFi.Ods.Features/OpenApiMetadata/OpenApiMetadataSdkGenTests.cs]: Removed

**Review by jamessmckay — APPROVED** [review]

**Pull request merged**